src/main/java/org/toradocu/extractor/JavadocExtractor.java

```diff
@@ -1,515 +1,664 @@
-
-      // Remove HTML tags (also generated by inline taglets). In the future, perhaps retain those
-      // tags, because they contain information that can be exploited.
-      comment = Jsoup.parse(comment).text();
-
-      // The list of the parameters of the method that we'll use in order to match
-      // the parameterName of the tag with the parameter itself
-      List<Parameter> parameters = getParameters(member);
-
-      String name = null; //Name of the ParamTag that we'll introduce
-      org.toradocu.extractor.Type type = null; //type of the parameter of the ParamTag
-      Boolean nullable = null; // Nullability of the parameter.
-
-      boolean found = false; //Boolean for control
-      for (int i = 0; !found && i < parameters.size(); i++) {
-        if (parameters.get(i).getName().equals(paramsTag.parameterName())) {
-          // If the tag parameterName matches any parameter name in the method, then we stop the
-          // loop.
-          found = true;
-          name = parameters.get(i).getName(); //then we assign values to the variables
-          type = parameters.get(i).getType();
-          nullable = parameters.get(i).getNullability();
+  /**
+   * Search for the type of the exception with the given type name. The type name is allowed to be
+   * fully-qualified or simple, in which case this method tries to guess the package name.
+   *
+   * @param classesInPackage list of class names in sourceMember's package
+   * @param sourceMember the source member for which the exception with type name {@code
+   *     exceptionTypeName} is expected
+   * @param exceptionTypeName the exception type name (can be fully-qualified or simple)
+   * @param className package of the class where {@code sourceMember} is defined
+   * @return the exception class
+   * @throws ClassNotFoundException if exception class couldn't be loaded
+   */
+  private Class<?> findExceptionType(
+      List<String> classesInPackage,
+      CallableDeclaration<?> sourceMember,
+      String exceptionTypeName,
+      String className)
+      throws ClassNotFoundException {
+    try {
+      return Reflection.getClass(exceptionTypeName);
+    } catch (ClassNotFoundException e) {
+      // Intentionally empty: Apply other heuristics to load the exception type.
+    }
+    // Try to load a nested class.
+    try {
+      return Reflection.getClass(className + "$" + exceptionTypeName);
+    } catch (ClassNotFoundException e) {
+      // Intentionally empty: Apply other heuristics to load the exception type.
+    }
+    try {
+      return Reflection.getClass("java.lang." + exceptionTypeName);
+    } catch (ClassNotFoundException e) {
+      // Intentionally empty: Apply other heuristics to load the exception type.
+    }
+    // Look in classes of package.
+    for (String classInPackage : classesInPackage) {
+      if (classInPackage.contains(exceptionTypeName)) {
+        // TODO Add a comment explaining why the following check is needed.
```

Comment:
Throughout this method, use data structures rather than error-prone, confusing string manipulation.